src/plots/plots.js

```diff
@@ -2848,7 +2849,50 @@ function _transition(gd, transitionOpts, opts) {
     return transitionStarting.then(function() { return gd; });
 }
 
+var typedArrays = {
+    int8: typeof Int8Array !== 'undefined' ? Int8Array : null,
+    uint8: typeof Uint8Array !== 'undefined' ? Uint8Array : null,
+    uint8clamped: typeof Uint8ClampedArray !== 'undefined' ? Uint8ClampedArray : null,
+    int16: typeof Int16Array !== 'undefined' ? Int16Array : null,
+    uint16: typeof Uint16Array !== 'undefined' ? Uint16Array : null,
+    int32: typeof Int32Array !== 'undefined' ? Int32Array : null,
+    uint32: typeof Uint32Array !== 'undefined' ? Uint32Array : null,
+    float32: typeof Float32Array !== 'undefined' ? Float32Array : null,
+    float64: typeof Float64Array !== 'undefined' ? Float64Array : null,
+    bigint64: typeof BigInt64Array !== 'undefined' ? BigInt64Array : null,
+    biguint64: typeof BigUint64Array !== 'undefined' ? BigUint64Array : null
+};
+
+function _decode(cont) {
+    if(cont.dtype && cont.bvals) {
+        var T = typedArrays[cont.dtype];
+        if(T) {
+            return new T(b64.decode(cont.bvals));
```

Comment:
@archmoj something like this?
```suggestion
            let arr = new T(b64.decode(cont.bvals));
            arr.shape = cont.shape;
            arr.dtype = cont.dtype;  // easier than arr.constructor.name 
```

Reply:
Or wrap it in an [ndarray](https://github.com/scijs/ndarray) object, which provides some more niceties, such as `arr.get(i, j)`.